packages/server/src/http/internals/HTTPResponse.tsx

```diff
@@ -0,0 +1,17 @@
+import { Dict } from '@trpc/server';
+import { URLSearchParams } from 'url';
+
+export type HTTPHeaders = Dict<string | string[]>;
```

Comment:
Any reason why this isn't `Record<string, string>` instead of `{ [x: string]: string | string[] | undefined; }`?

The output typing to api-gateway hates the fact that some of these headers can be undefined.

https://i.imgur.com/iXKBtLe.png

Reply:
Because it would be a lie and it's not really great practice to type anything as `Record<string, string>`

**Example:**

```ts
const obj: Record<string, string> = {
  foo: 'bar',
}


obj.bar.toLowerCase() // yay works
obj.nope.toLowerCase() // yay works (but it's actually a runtime error "Cannot read property 'toLowerCase' of undefined ")
```

Reply:
When you do the gateway adapter you'll have to deal with this yourself

Reply:
It's their lib that is wrong not to deal with that

Reply:
Interesting! Didn't realize how bad `Record<string, string>` could be.

Reply:
Yeah, that's why I have that `Dict`-helper, it's more accurate to the truth :) 

Simply cast the headers as `any` in your adapter maybe